### ISSUE TITLE
Treat 404 and other 'failures' as acceptable?

### ISSUE THREAD

**Issue by nitz** [thp/urlwatch#598]
This is mildly related to #352. Similarly, I have a page that is returning a 404, however the content of the page is actually valid, and I do want to filter/diff/report as usual. I tried using the `ignore_http_error_codes` directive, but that does as it says on the tin, just has it ignore it in my regularly scheduled `urlwatch` invocations.

I was hoping there'd be some way to specify `allow_http_response_codes` that would accept arguments like ignore does, but rather than ignoring, it treats the specified codes (or family of codes like `4xx`/`5xx`) as if they had been `200` and to continue the flow as usual.

I've got a workaround at the moment using `curl` as a shell job, but this seems like the sort of case that `urlwatch` would want to handle out of the box.

Wonderful tool, by the way, I'm very glad to have found it! ♥

**Comment by kongomongo** [comment]
Can't you replace the url job with a command job, and within that use a tool that ignores that 404  (for example curl -f)?

**Comment by nitz** [comment]
Yeah, that's what I'm currently using. It just feels to me that there is likely other cases besides my current use where a 404 (or other response) would be an acceptable HTTP response code that `urlwatch` should be able to handle out of the box. (Or even treat the HTTP response code as part of the diff from previous runs.)
